tango_simlib/tango_sim_generator.py

```diff
@@ -463,9 +461,9 @@ def get_device_class(sim_data_files):
     for data_file in sim_data_files:
         extension = os.path.splitext(data_file)[-1]
         extension = extension.lower()
-        if extension in [".xmi"]:
+        if extension in [".xmi", ".fgo"]:
             parser_instance = get_parser_instance(data_file)
-        elif extension in [".fgo", ".json"] and len(sim_data_files) < 2:
+        elif extension in [".json"] and len(sim_data_files) < 2:
             parser_instance = get_parser_instance(data_file)
```

Comment:
This loop looks like we are trying choose which file to parse using some kind of priority.  The order of preference appears to be:  xmi --> fgo --> json.  @kmadisa, can you recall the idea here?

If we are trying to do that, the code can be reworked to be a little clearer (replacing lines 459 to 468):
```
    precedence_map = {'.xmi': 1, '.fgo': 2, '.json': 3}

    def get_precedence(file_name):
        extension = os.path.splitext(file_name)[-1]
        extension = extension.lower()
        return precedence_map.get(extension, 100)

    sorted_files = sorted(sim_data_files, key=get_precedence)
    parser_instance = get_parser_instance(sorted_files[0])
```

Reply:
The idea was to get the class name from the xmi if at least two data files were used. But then there was a point at which we wanted to create a device simulator using the json file only and would get the class name from it, if and only if, it was the only data file used.
Let's go with your proposed changes. :+1: